### PR TITLE
added: swap event pair address

### PR DESCRIPTION
**Pull request by xjjda22**

### CODE REVIEW

File: components/BundleModal.tsx [hunk 1]
```diff
@@ -190,23 +190,26 @@ function BundleTransaction(transaction, index: number) {
   };
 
   useEffect( () => {
-    const getLogs = async () => setLogs(await getReceipts(transaction));
+    const getLogs = async () => {
+      const _logs = await getReceipts(transaction);
+      let _coins = [];
```
Comment (martriay) [inline components/BundleModal.tsx:195]: why destructuring the `reduce`?

File: components/BundleModal.tsx [hunk 2]
```diff
@@ -190,23 +190,26 @@ function BundleTransaction(transaction, index: number) {
...
+      await _logs.map( async (curr, i) => {
+        let { coin } = await curr;
+        if(coin.name != ""){
```
Comment (martriay) [inline components/BundleModal.tsx:198]: please use `===` and `!==`

File: components/BundleModal.tsx [hunk 3]
```diff
@@ -238,19 +241,23 @@ function BundleTransaction(transaction, index: number) {
...
+                    coins[coin].logo
+                      ? <img className="w-4 mr-1" key={ "i_" + index + now() } src={coins[coin].logo} />
+                      : <></>
+                  }
+                  <b>{ coin }</b>
+                </a>
+                <span style={{ margin: 2}}>
+                  { coins[coin].value > 0 ? ` ${coins[coin].value}` : "" }
+                  { coins[coin].ethValue > 0 ? ` ($${coins[coin].ethValue})` : "" }
+                  { coins[coin].event ? ` (${coins[coin].event})` : ""  }
```
Comment (martriay) [inline components/BundleModal.tsx]: As mentioned in https://github.com/martriay/flashbots-explorer/pull/17#discussion_r649367680,  please show event name only when no value has been transferred.

File: lib/ABILogs.js [hunk 4]
```diff
@@ -38,10 +38,14 @@ addEvents();
...
-    const { coin, logo, decimals } = getToken(log.address);
+  return await decodeLogs(_logs).map(async log => {
+    let { coin, logo, decimals } = await getToken(log.address);
+    if(coin == "") {
+      let pair = await getPairAddress(log.address);
+      coin = pair.coin;
+    }
     let ethValue = 0;
-    let value;
+    let value = 1;
```
Comment (martriay) [inline lib/ABILogs.js]: It's important not to hardcode a value here if there's no value being transferred.

File: lib/getReceipts.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ export async function getReceipts(transaction) {
   try {
     const res = await fetch(`https://mainnet.infura.io/v3/${process.env.NEXT_PUBLIC_INFURA_ID}`, params);
     const { result : { logs } } = await res.json();
-    return getAllLogs(logs);
+    return await getAllLogs(logs);
```
Comment (martriay) [inline lib/getReceipts.js]: why not `Promise.all()`?

File: lib/uniswapV2SubGraph.ts [hunk 6]
```diff
@@ -32,3 +46,31 @@ export async function getEthPrice() {
...
+      method: 'POST',
+      headers: {
+        'Accept': 'api_version=2',
+        'Content-Type': 'application/graphql'
+      },
+      body: JSON.stringify({ query : gQL })
+    });
+
+    let { data: { pair } } = await res.json();
+    if(pair != null){
```
Comment (martriay) [inline lib/uniswapV2SubGraph.ts]: Same as above, `===` and `!==` are preferred

File: lib/ABILogs.js [hunk 7]
```diff
@@ -38,10 +38,14 @@ addEvents();
 
 export async function getAllLogs(_logs) {
   const ethPrice = await getEthPrice();
-  return decodeLogs(_logs).map(log => {
-    const { coin, logo, decimals } = getToken(log.address);
+  return await decodeLogs(_logs).map(async log => {
+    let { coin, logo, decimals } = await getToken(log.address);
+    if(coin == "") {
```
Comment (martriay) [inline lib/ABILogs.js]: Same as above

File: lib/uniswapV2SubGraph.ts [hunk 8]
```diff
@@ -32,3 +46,31 @@ export async function getEthPrice() {
...
+      c.coin = `${pair.token0.symbol}-${pair.token1.symbol}`
```
Comment (martriay) [inline lib/uniswapV2SubGraph.ts:72]: what about making it an array so it's easier and flexible to handle it from the components?

**Review by martriay — CHANGES_REQUESTED** [review]
Cool, thanks once more! Left some comments.

File: components/BundleModal.tsx (hunk 1 above)
Comment (xjjda22) [reply components/BundleModal.tsx:195]: change to map becuz reduce was throwing error

File: lib/uniswapV2SubGraph.ts (hunk 8 above)
Comment (xjjda22) [reply lib/uniswapV2SubGraph.ts:72]: if multiple then can use array otherwise object

**Comment by martriay** [comment]
why do you open and close it?

**Pull request closed without merging**